commands/addrole.js

```diff
@@ -0,0 +1,127 @@
+const mongoose = require('mongoose');
+const conf = require('../conf.js');
+mongoose.connect(conf.mongodbs, { useNewUrlParser: true })
+
+/**
+ * Adds a new role to any message in the server
+ */
+module.exports = async (bot, message, args) => {
+    // check for help
+    if (!args[2] || args[0] == "help") {
+        let help = new Discord.RichEmbed()
+            .setColor("#FFFF00")
+            .addField("Usage", "addemo <message_id> <emote> <role>")
+            .addField("More info", "Adds a new role to any message in the server.")
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    // check if command issuer has perms
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
+            await message.delete()
+            return m.delete(15000)
+        })
+    }
+    // check if bot has perms
+    if (!message.guild.me.hasPermission("MANAGE_ROLES")) {
+        noperm = new Discord.RichEmbed().setColor("#FF0000").addField("Error", "The bot currently doesn't have permission to add or remove roles from people, please give the bot permission before adding reaction roles on messages.")
+        return message.reply(noperm).then(r => r.delete(30000))
+    }
+    // set variables for easier access
+    var message_id = args[0]
+    var emoji = args[1]
+    var role = message.mentions.roles.first()
+    var channel_id
+    // message placeholder
+    var msg
+    // RichEmbed placeholder
+    let clean = new Discord.RichEmbed()
+    // start looking for the message with the specified id, in order to react to it with the added emoji
+    let channels = message.guild.channels
+    channels = channels.first(channels.array().length)
+    // channels is now an arrray of all the channels in this server
+    let l = channels.length
+    let i
+    // now keep looping through the text channels and try to fetch the message
+    for (i = 0; i < l; i++) {
+        if (channels[i].type == "text") {
+            good_message = await channels[i].fetchMessage(message_id).catch(err => { })
+            // once message is found, react to it with the given emoji
+            if (good_message) {
+                await good_message.react(emoji).catch(err => { })
+                channel_id = channels[i].id
+            }
+        }
+    }
+    // if no role was mentioned in the command call
+    if (!role) {
+        return message.channel.send(
+            clean
+                .setColor("#FF0000")
+                .addField("Error", "No roles were mentioned in your command, please mention the role to add.")
+        ).then(r => r.delete(30000))
+    }
+    // check if there's an emoji already bound to that message with a role.
+    reaction_message_model.findOne({
+        messageID: message_id
+    }, async (err, reaction_message) => {
+        msg = await message.reply("Loading...")
+        if (err) console.log(err)
```

Comment:
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
	if (err)
		console.log(err)
```